week2/The_system_of_the_student_ score_management_v1.0.cpp

```diff
@@ -0,0 +1,179 @@
+#include <stdio.h>
+
+void input_score(long student_ID[], int score[], int n);
+void print_total_and_average(int n, int score[]);
+void rank_by_grade_in_descending_order(long student_ID[], int score[], int n);
+void print_score(long student_ID[], int score[], int n);
+void rank_by_student_ID_in_ascending_order(long student_ID[], int score[], int n);
+int search_student_rank_and_grade_by_student_ID(long student_ID[], int score[], int n);
+void count_the_percentage_of_the_each_category(long student_ID[], int score[], int n);
+
+int main()
+{
+	long student_ID[30];
+	int score[30], n, choice;
+	printf("1. Input record\n");
+	printf("2. Caculate total and average score of course\n");
+	printf("3. Sort in descending order by score\n");
+	printf("4. Sort in ascending order by number\n");
+	printf("5. search by number\n");
+	printf("6. Statistic analysis\n");
+	printf("0. Exit\n\n");
+	do {
+		printf("Please enter your choice:");
+		scanf_s("%d", &choice);
+		if (choice <= 6 && choice >= 0) {
+			switch (choice)
+			{
+			case 1:
+				printf("Please input the number of the students(n <= 30):");
+				scanf_s("%d", &n);
+				input_score(student_ID, score, n);
+				break;
+			case 2:
+				print_total_and_average(n, score);
+				break;
+			case 3:
+				rank_by_grade_in_descending_order(student_ID, score, n);
+				printf("Rank by grade in descending order\n");
+				print_score(student_ID, score, n);
+				break;
+			case 4:
+				rank_by_student_ID_in_ascending_order(student_ID, score, n);
+				printf("rank by student number in ascending order\n");
+				print_score(student_ID, score, n);
+				break;
+			case 5:
+				search_student_rank_and_grade_by_student_ID(student_ID, score, n);
+				break;
+			case 6:
+				count_the_percentage_of_the_each_category(student_ID, score, n);
+				break;
+			case 0:
+				printf("Exit successfully! \\*_*/");
+				break;
+			}
+		}
+		else {
+			printf("Please input a correct choice.  !!-_-!!\n");
+		}
+	} while (choice != 0);
+	return 0;
+}
+
+void input_score(long student_ID[], int score[], int n) {
+	if (n > 30) {
+		printf("The number exceeds limit.\n");
+	}
+	else {
+		printf("Please input the student's number and his score\n");

```

Comment:
What if the student's gender is female? Keep diversity in mind during development.